### PR TITLE
accounts/abi: simulator implements interface PendingStateReader  

### PR DESCRIPTION
**Pull request by jsvisa**
Implement the first interface mentioned in https://github.com/ethereum/go-ethereum/issues/28431

### CODE REVIEW

File: accounts/abi/bind/backends/simulated.go [hunk 1]
```diff
@@ -507,15 +499,52 @@ func (b *SimulatedBackend) PendingCallContract(ctx context.Context, call ethereu
...
+	return b.pendingState.GetOrNewStateObject(account).Balance(), nil
+}
+
+// PendingStorageAt implements PendingStateReader.PendingStorageAt, retrieving
+// the account's storage value of key in the pending state.
+func (b *SimulatedBackend) PendingStorageAt(ctx context.Context, account common.Address, key common.Hash) ([]byte, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	return b.pendingState.GetOrNewStateObject(account).Code(), nil
```
Comment (MariusVanDerWijden) [inline accounts/abi/bind/backends/simulated.go:517]: This looks wrong to me

Comment (jsvisa) [reply accounts/abi/bind/backends/simulated.go]: sorry, by mistake.

**Comment by s1na** [comment]
I know I raised the ticket you are fixing but my goal was to discuss it first:
- One argument for not adding these to simulated backend is that there's a rough consensus that we will drop the pending state from geth. Which means these methods will have to be removed possibly soon after they are added.
- Also people can omit the PendingStateReader from the interface and they would be able to switch between ethclient and simulated backend seemlessly.
My opinion is it's fine to add these methods until we remove pending state. Open to hear other opinions.

**Comment by fjl** [comment]
Closing because we have another PR that rewrites the simulated backend based on ethclient. That version will have the methods automatically.

**Comment by fjl** [comment]
This is the PR: https://github.com/ethereum/go-ethereum/pull/28202

**Pull request closed without merging**